packages/autorest.python/autorest/codegen/serializers/builder_serializer.py

```diff
@@ -815,72 +810,75 @@ def _initialize_overloads(
-                    (
-                        o
-                        for o in builder.overloads
-                        if isinstance(o.parameters.body_parameter.type, BinaryType)
-                    )
-                ),
-            )
-            binary_body_param = binary_overload.parameters.body_parameter
-            retval.append(
-                f"if {binary_body_param.type.instance_check_template.format(binary_body_param.client_name)}:"
-            )
-            if binary_body_param.default_content_type and not same_content_type:
-                retval.append(
-                    f'    content_type = content_type or "{binary_body_param.default_content_type}"'
-                )
-            retval.extend(
-                f"    {l}" for l in self._create_body_parameter(binary_overload)
-            )
-            retval.append("else:")
-            other_overload = cast(
-                OperationType,
-                next(
-                    (
-                        o
-                        for o in builder.overloads
-                        if not isinstance(o.parameters.body_parameter.type, BinaryType)
-                    )
-                ),
-            )
-            retval.extend(
-                f"    {l}" for l in self._create_body_parameter(other_overload)
-            )
-            if (
-                other_overload.parameters.body_parameter.default_content_type
-                and not same_content_type
-            ):
-                retval.append(
-                    "    content_type = content_type or "
-                    f'"{other_overload.parameters.body_parameter.default_content_type}"'
-                )
-        except StopIteration:
-            for idx, overload in enumerate(builder.overloads):
-                if_statement = "if" if idx == 0 else "elif"
+            overload_retval: List[str] = []
+            client_names = [
+                overload.request_builder.parameters.body_parameter.client_name
+                for overload in builder.overloads
+            ]
+            for v in sorted(set(client_names), key=client_names.index):
+                overload_retval.append(f"_{v}: Any = None")
+
+            # make sure some special type is in last position and some in first position
+            # but we can't change original data
+            overloads_copy = [
+                o
+                for o in builder.overloads
+                if o.parameters.body_parameter.type.enable_overload_check
+            ]
+            for i, _ in enumerate(overloads_copy):
+                if isinstance(
+                    overloads_copy[i].parameters.body_parameter.type,
+                    (AnyType, AnyObjectType),
+                ):
+                    _swap(overloads_copy, i, -1)
+                if isinstance(
+                    overloads_copy[i].parameters.body_parameter.type, BinaryType
+                ):
+                    _swap(overloads_copy, i, 0)
```

Comment:
i really don't like moving overloads around in operations once they're already constructed. i agree we want to put higher visibility overloads, but this is somethign we should be doing in preprocess, not once they're already created

Reply:
The logic here is only for check order of main function and doesn't influence order of @overload function. 
In actually, we can do it preprocess or serializer. If in preprocess, the order of @overload function will also be influenced that makes the PR more messy.

Reply:
i'm confused: what is this swap doing if it doesn't influence theorder of the overload function? also why would movingg this into preprocess make this pr messier?

Reply:
we render the overload function with order: str, bytes, IO. But we check instance with order bytes/IO, str. How do we do it? The answer is swap the order for check logic
https://github.com/Azure/autorest.python/blob/00bc12a382db3cd3cd33b97c5c815c8a347312d2/packages/typespec-python/test/generated/overload/overload/_operations/_operations.py#L73-L153